### PR TITLE
Push Notifications: Add Tracks events

### PR DESCRIPTION
**Pull request by jonathansadowski**
Moved from #5993 in order to make this PR against master.

Test live: https://calypso.live/?branch=add/push-notification/track

### CODE REVIEW

**Comment by jonathansadowski** [comment]
I made a change to address the failing tests -- so I marked this as "Needs Review" again

File: client/lib/service-worker/service-worker.js [hunk 1]
```diff
@@ -35,25 +35,35 @@ self.addEventListener( 'push', function( event ) {
 		self.registration.showNotification( notification.msg, {
 			tag: 'note_' + notification.note_id,
 			icon: notification.icon,
-			timestamp: notification.note_timestamp
+			timestamp: notification.note_timestamp,
+			data: notification
+		} ).then( function () {
+			if ( notification.note_opened_pixel ) {
+				fetch( notification.note_opened_pixel, { mode: 'no-cors' } ).catch( function( err ) {
+					console.log( 'Could not load the pixel %s', notification.note_opened_pixel );
```
Comment (jblz) [inline client/lib/service-worker/service-worker.js]: Where do messages go when we call `console.log` in the service worker?

Comment (jonathansadowski) [inline client/lib/service-worker/service-worker.js]: In Chrome, the service worker has its own console, which can be accessed via `chrome://serviceworker-internals/`.
In Firefox, the browser console (`Shift` + `Command` + `J`)

File: client/state/push-notifications/actions.js [hunk 2]
```diff
@@ -37,12 +37,20 @@ import {
...
+import {
+	recordTracksEvent
+} from 'state/analytics/actions';
 
 const debug = debugFactory( 'calypso:push-notifications' );
 const DAYS_BEFORE_FORCING_REGISTRATION_REFRESH = 15;
 const serviceWorkerOptions = {
 	path: '/service-worker.js',
 };
+const tracksEvent = ( dispatch, eventName, props ) => {
```
Comment (jblz) [inline client/state/push-notifications/actions.js]: If you look at what ultimately [gets executed](https://github.com/Automattic/wp-calypso/blob/673fd7cced1d5c8ff75ce47ecf3cf24b8c279455/client/lib/analytics/index.js#L122-L147), it doesn't block for very long, so the timeout is a pretty small optimization here, IMO.

I'd suggest if we are going to use the timeout, I'd add an action called `asyncRecordTracksEvent` to the [analytics actions](https://github.com/Automattic/wp-calypso/blob/673fd7cced1d5c8ff75ce47ecf3cf24b8c279455/client/state/analytics/actions.js#L58) module to keep from passing `dispatch` around.

File: client/lib/service-worker/service-worker.js (hunk 1 above)
Comment (jblz) [inline client/lib/service-worker/service-worker.js]: Cool. Leaving it in works for me, then.

File: client/state/push-notifications/actions.js (hunk 2 above)
Comment (jonathansadowski) [inline client/state/push-notifications/actions.js]: I agree, I've updated and I've remove the timeout

**Comment by jblz** [comment]
Works well. 👍

**Comment by jblz** [comment]
@jonathansadowski I ran the timing on that dispatch call... btw. 2-3ms w/o the `setTimeout` -- in the 100ths of a ms range w/ it, so possibly worth adding in at some point, but I'd either add it as an action to the analytics module as mentioned, or just call `setTimeout` around each dispatch.

**Pull request merged**